doc/xversionmessage.md

```diff
@@ -0,0 +1,250 @@
+without having ever send an `xversion` message should be handled as a
+protocol violation.
+
+An empty `xmap` for a peer has a defined reading (see below).
+However, to simplify node implementations, it is deemed acceptable to
+enable certain protocol features only after proper receipt of a
+corresponding `xversion` message.
+
+## Interpretation of the `xmap`
+
+If a key cannot be found in the `xmap`, its value must be assumed to be an
+empty byte vector. An entry that has unknown meaning is to be ignored.
+
+If further structuring of a byte-vector value is desired, it should be
+done using the standard Bitcoin de-/serialization format.
+
+For obvious reasons, the enumeration of entries and their expected
+types in the map should be agreed upon between implementations. For
+this reason, a separate specification that lists configuration and
+version keys is introduced, named the `xmap directory` (see below).
+
+The format of `std::vector<uint8_t>` is unwieldy for configuration and
+version parameters that could be expressed as simple integer to
+integer mappings. For this reason, a specific interpretation for
+certain value types is assumed. The list of these value types is
+expected to grow over time and this document correspondingly
+extended. For now, only a compact-size encoded unsigned 64-bit integer
+value named `u64c` is specified.
+
+### The `u64c` value type
+
+Values in the `xmap` can be interpreted as compact-size encoded 64-bit
+unsigned integers (named `u64c`), allowing to use it as a simple
+unsigned to unsigned integer map.
+
+A value of `type` is encoded in Bitcoin's message serialization as a
+COMPACT_SIZE integer *within* the generic byte vector value of the `xmap`.
+For example, a value of `0x1000` (4096) is encoded like this as the `xmap`
+value:
+`FD 00 10`
+
+Which gets further encoded in the `xmap` encoding itself as:
+`03 FD 00 10`
+
+A special exception is added for empty value vectors. Empty value
+vectors are to be interpreted as value zero in the `u64c`
+reading. Thus missing entries in the `xmap` are interpreted as `u64c`
+zero values.
+
+When interpreting these `uint64_t` values from the table and certain
+bits of the value are unused for a given key, the implementer should
+mask out the needed bits using an AND-mask to allow for future use of
+the yet unused bits in a given value.
+
+If decoding as a compact size integer fails, a value of zero for
+the corresponding value should be assumed.
+
+## The `xmap` directory
+
+A directory that lists the currently defined key values comes along
+with this specification and can be found in the file named
+`xversionkeys.dat`.
+
+At the moment, and for more compact wire encoding, keys are assumed to
+be just 32-bit in size and the current specification format reflects
+this. They are assumed to consist of a 16-bit prefix in the more
+significant bits of the 64-bit integer (bits 16..31) followed by a
+16-bit suffix in the least significant bits (0..15).
```

Comment:
My idea was to refrain from doing this until there's a need to do so - and rethink *at that time* whether to change anything about the process.

You can use any value in a straight-forward way *right now*. I just thought it would make sense to not do this. The longer keys might also never be needed.